### PR TITLE
chore(component-testing): move routes-ct to TS

### PR DESCRIPTION
**Pull request by lmiller1990**

### CODE REVIEW

File: packages/server-ct/index.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  invokedFromCli: boolean
+  config: Record<string, unknown>
+  projectRoot: string
+  browsers: Cypress.Browser[]
+  proxyUrl: string
+  userAgent: boolean
+  socketIoRoute: string
+  chromeWebSecurity: boolean
+  proxyServer?: string
+}
```
Comment (JessicaSachs) [inline packages/server-ct/index.ts:17]: I believe the CLI options are typed elsewhere... @flotwig said to check
`cli/types/cypress` or `cli/types/npm-module`

Can we use existing types?

File: packages/server-ct/src/routes-ct.ts [hunk 2]
```diff
@@ -58,12 +68,14 @@ module.exports = ({ app, config, project, onError }) => {
   // during routing just log them out to
   // the console and send 500 status
   // and report to raygun (in production)
-  app.use((err, req, res) => {
+  const errorHandlingMiddleware: ErrorRequestHandler = (err, req, res, next) => {
```
Comment (JessicaSachs) [inline packages/server-ct/src/routes-ct.ts:71]: Unused next parameter? Is there a reason typescript needs you to pass this through?
```suggestion
  const errorHandlingMiddleware: ErrorRequestHandler = (err, req, res) => {
```

File: packages/server-ct/src/server-ct.ts [hunk 3]
```diff
@@ -58,8 +59,8 @@ export class Server {
     return app
   }
 
-  createRoutes (...args) {
-    return require('./routes-ct').apply(null, args)
+  createRoutes (...args: unknown[]) {
```
Comment (JessicaSachs) [inline packages/server-ct/src/server-ct.ts:62]: Can we properly type these args? There's only one usage of `createRoutes`. The `app` and `config` and `project` are passed in inside of `Sever#open`... and have fairly defined business domain concepts

Comment (JessicaSachs) [reply packages/server-ct/src/server-ct.ts:62]: In general, `config` is referenced throughout this file, and I'd love to get type completion for it. We use the config object more than anything to grab data from it and I always forget what's on the schema.

**Comment by lmiller1990** [comment]
Closed in favor of https://github.com/cypress-io/cypress/pull/9737

**Review by JessicaSachs — CHANGES_REQUESTED** [review]
Can we add additional typings in this file? I would really like to see `config` and some of the options patterns typed.

Lastly, I think some of the Types are already defined for `OpenConfig` -- but I may be mistaken.

**Pull request closed without merging**